prymer/model.py

```diff
@@ -83,6 +83,55 @@ def __str__(self) -> str:
         return f"(min:{self.min}, opt:{self.opt}, max:{self.max})"
 
 
+@dataclass(slots=True, frozen=True, init=True)
+class WeightRange(Generic[Numeric]):
```

Comment:
I like this, but a few thoughts:

- It's not really a range, more of a pair.  I.e. I can have a higher penalty weight for "primer tm too low" than "primer tm too high" - they're independent values.  Perhaps it's just `Weights`?
- I think in p3 weights are always floats because they are scaling factors, so i think we could un-generify it and juse always have the values be floats